File: src/1_two_sum/Solution.java
import java.util.Map;
import java.util.HashMap;

public class A1_two_sum {

    public static int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++)
        {
            if(map.containsKey(target - nums[i])) {
                result[0] = i ;
                result[1] = map.get(target - nums[i]);
                return result;
            }
            map.put(nums[i], i);
        }
        return result;
    }

    public static void main (String[] args){
        int[] nums = new int[5];
        nums[0] = 3;
        nums[1] = 4;
        nums[2] = 9;
        nums[3] = 6;
        nums[4] = 7;
        int[] result = twoSum(nums , 9);
        System.out.println(result[0]);
        System.out.println(result[1]);
    }

}

File: src/1_two_sum/solution.py
from typing import List

class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        print(nums)
        print(target)
        for i,n in enumerate(nums):
            m = target - n
            if m in d :
                return [d[m], i]
            d[n] = i
        return

    # init method or constructor
    def __init__(self, name):
        self.name = name 

p = TwoSum('PNKC')
target = 15
nums = [3, 2, 12, 14, 13]
result = p.twoSum(nums, target)
print( result)

